spot_driver/spot_driver/calibrated_reregistered_hand_camera_depth_publisher.py

```diff
@@ -0,0 +1,254 @@
+# Copyreference (c) 2024 Boston Dynamics AI Institute LLC. All references reserved.
+
+"""
+This script exists as it is not currently possible to set the intrinsic and extrinsic
+parameters for the hand cameras through the bosdyn API. Currently, the registered
+depth image is published using extrinsic and intrinsic parameters that are set under the hood. 
+If the updated results from the calibration in spot_wrapper/spot_wrapper/calibration
+can be populated into the underlying bosdyn API, then the default registered depth image
+could be used without this script.
+
+While it is possible to calibrate the gripper camera entirely under the hood, there is no way to
+only set the intrinsic/extrinsic parameters (to preserve control over calibration parameters)
+
+This script assumes you have a calibration generated with /spot_wrapper/spot_wrapper/calibration
+"""
+import argparse
+from typing import Optional
+
+import bdai_ros2_wrappers.process as ros_process
+import bdai_ros2_wrappers.scope as ros_scope
+import cv2
+import numpy as np
+import open3d as o3d
+import yaml
+from bdai_ros2_wrappers.context import wait_for_shutdown
+from cv_bridge import CvBridge
+from sensor_msgs.msg import Image
+
+
+class CalibratedReRegisteredHandCameraDepthPublisher:
+    def __init__(
+        self,
+        calibration_path: Optional[str] = None,
+        tag: str = "default",
+        robot_name: str = "spot",
```

Comment:
Along with my other comment, I think it would make sense for this to be either `""` or `Optional[str] = None`